packages/commercetools/api-client/src/helpers/cart/actions.ts

```diff
@@ -22,34 +23,39 @@ export const createChangeLineItemQuantityAction = (product: LineItem) => ({
   }
 });
 
-export const setShippingAddressAction = (shippingDetails: Address): { setShippingAddress: { address: AddressInput } } => ({
-  setShippingAddress: {
-    address: {
-      title: shippingDetails.title,
-      salutation: shippingDetails.salutation,
-      firstName: shippingDetails.firstName,
-      lastName: shippingDetails.lastName,
-      streetName: shippingDetails.streetName,
-      streetNumber: shippingDetails.streetNumber,
-      additionalStreetInfo: shippingDetails.additionalStreetInfo,
-      postalCode: shippingDetails.postalCode,
-      city: shippingDetails.city,
-      region: shippingDetails.region,
-      state: shippingDetails.state,
-      country: shippingDetails.country,
-      company: shippingDetails.company,
-      department: shippingDetails.department,
-      building: shippingDetails.building,
-      apartment: shippingDetails.apartment,
-      pOBox: shippingDetails.pOBox,
-      phone: shippingDetails.contactInfo?.phone,
-      mobile: shippingDetails.contactInfo?.mobile,
-      email: shippingDetails.contactInfo?.email,
-      fax: shippingDetails.contactInfo?.fax,
-      additionalAddressInfo: shippingDetails.additionalAddressInfo
-    }
+export const setShippingAddressAction = (shippingDetails: Address): { setShippingAddress: { address: AddressInput } } => {
+  if (shippingDetails.contactInfo?.phone || shippingDetails.contactInfo?.email || shippingDetails.contactInfo?.mobile || shippingDetails.contactInfo?.fax) {
```

Comment:
```
const hasContactInfo = details => Object.keys(details.contactInfo).some(c => ['phone', 'email', 'mobile', 'fax'].includes(c))
```

Reply:
and use it in both places

Reply:
ok i wasnt sure if you want to keep support for contactInfo or ditch it so did not optimize at all ;]